### DISCUSSION TITLE
Include existing function polyfills in the installer

### DISCUSSION THREAD

**Discussion by GravisZro** [itsmattkc/dotnet9x#37] [Ideas]
[Open Layer for Unicode](https://sourceforge.net/projects/opencow/) A.K.A. Opencow replaces all the *W function stubs and implements the missing ones.
[KernelEx](https://sourceforge.net/projects/kernelex/) implements several kernel functions present in the NT kernel series that is missing from the 9x series.

Adding these two to the installer will definitely boost compatibility for applications, .Net and otherwise.

Does anyone know of any other compatibility layers like these?

**Comment by jamsilva** [comment]
There's some [extensions for KernelEx that add additional APIs](https://msfn.org/board/topic/157173-kext-diy-kernelex-extensions/) and it should be possible to port to 95 and extend it with additional kernel-side APIs since it has its own VxD.
Opencow seems to have a simpler method that could be extended as well. For example [here's how it deals with CreateFileMappingW](https://github.com/clamwin/opencow/blob/e4b101916d538025ab5be42b4fd33669610c6dd6/src/kernel32.cpp#L1101) mentioned in #11.

**Comment by GravisZro** [comment]
Sadly, the authors didn't add to the KernelEx project or post their source in any public repos.  However, there seems to be some preserved bits in an archive.org backup of the now unavailable files.: https://archive.org/details/msfn-archived-topics-june-2021.-7z_202106
